src/main/java/org/mule/weave/v2/jwt/RSAHelper.java

```diff
@@ -38,15 +35,45 @@ public static String signString(String content, String privateKeyContent, String
         return Base64.getUrlEncoder().encodeToString(s).replace("=", "");
     }
 
-    private static PrivateKey getPrivatePKCS1Pem(String privateKeyContent) throws IOException, NoSuchAlgorithmException, InvalidKeySpecException {
+    public static String signStringWithKeyStore(String content, String keystorePath, String keystoreType, String keystorePassword, String keyAlias, String algorithm) throws NoSuchAlgorithmException, IOException, KeyStoreException, CertificateException, UnrecoverableKeyException, InvalidKeyException, SignatureException {
+        PrivateKey pk = getPrivatePKCS8Pem(keystorePath, keystoreType, keystorePassword, keyAlias);
+        Signature privateSignature = Signature.getInstance(algorithm);
+        privateSignature.initSign(pk);
+        privateSignature.update(content.getBytes(StandardCharsets.UTF_8));
+        byte[] s = privateSignature.sign();
+        return Base64.getUrlEncoder().encodeToString(s).replace("=", "");
+    }
+
+    private static PrivateKey getPrivatePKCS1Pem(String privateKeyContent) throws NoSuchAlgorithmException, InvalidKeySpecException {
         privateKeyContent = privateKeyContent.replaceAll("\\n", "").replace(BEGIN_RSA_KEY, "")
                 .replace(END_RSA_KEY, "");
         byte[] bytes = Base64.getDecoder().decode(privateKeyContent);
 
         RSAPrivateCrtKeySpec keySpec = decodeRSAPrivatePKCS1(bytes);
         KeyFactory keyFactory = KeyFactory.getInstance("RSA");
-        PrivateKey privateKey = keyFactory.generatePrivate(keySpec);
-        return privateKey;
+        return keyFactory.generatePrivate(keySpec);
+    }
+
+    private static PrivateKey getPrivatePKCS8Pem(String keystorePath, String keystoreType, String keystorePassword, String keyAlias) throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException, UnrecoverableKeyException {
+        KeyStore keystore = KeyStore.getInstance(keystoreType);
+
+        InputStream is = RSAHelper.class.getClassLoader().getResourceAsStream(keystorePath);
+
+        keystore.load(is, keystorePassword.toCharArray());
```

Comment:
We should check that the keystorePath is valid if not this will throw NPE

Reply:
So throw `IllegalStateException` if the input stream is null or the path is empty?

Reply:
@machaval Are there some docs with examples on how to write negative/error test cases?